ui/app/src/components/BrokenReferencesDialog/BrokenReferencesDialogContainer.tsx

```diff
@@ -32,20 +31,52 @@ import DialogFooter from '../DialogFooter';
 import SecondaryButton from '../SecondaryButton';
 import PrimaryButton from '../PrimaryButton';
 import ApiResponseErrorState from '../ApiResponseErrorState';
+import { fetchDependant } from '../../services/dependencies';
+import type { LegacyItem } from '../../models';
+import { parseLegacyItemToContentItem } from '../../utils/content';
+import { pushDialog } from '../../state/actions/dialogStack';
 
 export function BrokenReferencesDialogContainer(props: BrokenReferencesDialogContainerProps) {
-	const { references, error, onClose, onContinue } = props;
+	const { path, references: initialReferences, error, onClose, onContinue } = props;
 	const dispatch = useDispatch();
 	const site = useActiveSiteId();
 	const { authoringBase } = useEnv();
+	const [references, setReferences] = useState(initialReferences || []);
 
 	const onContinueClick = (e) => {
 		onClose(e, null);
 		onContinue();
 	};
 
-	const onEditReferenceClick = (path: string) => {
-		dispatch(showEditDialog({ path, authoringBase, site, onSaveSuccess: fetchBrokenReferences() }));
+	const onEditReferenceClick = (referencePath: string) => {
+		dispatch(
+			pushDialog({
+				component: 'craftercms.components.LegacyFormDialog',
```

Comment:
Anywhere showing FE should be using the current util that shows either form based on the storage and defaults to FE.

Please review if you're opening legacy FE directly elsewhere anywhere on the PR.